File: web-service/yacontest-helper/src/main/java/org/ilyatyamin/yacontesthelper/scheduling/SchedulingConfig.java
package org.ilyatyamin.yacontesthelper.scheduling;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;

@Configuration
public class SchedulingConfig {
    private static final int THREAD_POOL_SIZE = 5;

    @Bean
    public ThreadPoolTaskScheduler updateGoogleSheetsScheduler(){
        ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
        threadPoolTaskScheduler.setPoolSize(THREAD_POOL_SIZE);
        threadPoolTaskScheduler.setThreadNamePrefix("updateGoogleSheetsScheduler");
        return threadPoolTaskScheduler;
    }
}

File: web-service/yacontest-helper/src/main/java/org/ilyatyamin/yacontesthelper/configs/SchedulingConfig.kt
package org.ilyatyamin.yacontesthelper.configs

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler

@Configuration
open class SchedulingConfig {
    @Bean
    open fun updateGoogleSheetsScheduler(): ThreadPoolTaskScheduler {
        val threadPoolTaskScheduler = ThreadPoolTaskScheduler()
        threadPoolTaskScheduler.poolSize = THREAD_POOL_SIZE
        threadPoolTaskScheduler.threadNamePrefix = "updateGoogleSheetsScheduler"
        return threadPoolTaskScheduler
    }

    companion object {
        private const val THREAD_POOL_SIZE = 5
    }
}
